workspace-engine/src/contract.rs

```diff
@@ -0,0 +1,317 @@
+use crate::operation::{
+    CallCall, CallDeployCode, CallDeployErc20Token, CallDeployUpgrade, CallDeposit,
+    CallFactorySetWNearAddress, CallFactoryUpdate, CallFactoryUpdateAddressVersion,
+    CallFtOnTransfer, CallFtTransfer, CallFtTransferCall, CallPausePrecompiles, CallRefundOnError,
+    CallRegisterRelayer, CallResumePrecompiles, CallSetEthConnectorContractData, CallStageUpgrade,
+    CallStateMigration, CallStorageDeposit, CallStorageUnregister, CallStorageWithdraw, CallSubmit,
+    CallWithdraw, ViewBalance, ViewBlockHash, ViewBridgeProver, ViewChainId, ViewCode,
+    ViewFtBalanceOf, ViewFtBalanceOfEth, ViewFtMetadata, ViewFtTotalEthSupplyOnAurora,
+    ViewFtTotalSupply, ViewIsUsedProof, ViewNonce, ViewOwner, ViewPausedPrecompiles, ViewStorageAt,
+    ViewStorageBalanceOf, ViewUpgradeIndex, ViewVersion, ViewView,
+};
+use crate::types::Account;
+use aurora_engine::fungible_token::FungibleTokenMetadata;
+use aurora_engine_types::types::{Address, Balance};
+use aurora_engine_types::U256;
+use aurora_workspace_types::input::ProofInput;
+use aurora_workspace_types::{AccountId, H256};
+use aurora_workspace_utils::Contract;
+#[cfg(feature = "ethabi")]
+use ethabi::{ParamType, Token};
+use near_sdk::json_types::U128;
+use serde_json::json;
+
+#[derive(Debug, Clone)]
+pub struct EngineContract {
+    contract: Contract,
+}
+
+impl EngineContract {
+    pub fn new(contract: Contract) -> Self {
+        Self { contract }
+    }
+
+    pub fn as_contract(&self) -> &Contract {
+        &self.contract
+    }
+
+    pub fn id(&self) -> &AccountId {
+        self.contract.id()
+    }
+
+    pub fn new_from_contract(contract_id: AccountId, account: Account) -> Self {
+        Self {
+            contract: Contract::new(contract_id, account),
+        }
+    }
+}
+
+/// Callable functions
+impl EngineContract {
```

Comment:
Don't you think to create a trait with these methods?

Reply:
We decided to do that a bit later.